packages/driver/test/cypress/integration/commands/navigation_spec.coffee

```diff
@@ -567,6 +567,15 @@ describe "src/cy/commands/navigation", ->
               "http://localhost:3500/fixtures/dimensions.html"
             ])
 
+    ## https://github.com/cypress-io/cypress/issues/1311
+    it "doesn't invoke onLoad callback when visiting the same URL with hashes", ->
+      cy
+        .visit("http://localhost:3500/fixtures/generic.html#foo")
+        .visit("http://localhost:3500/fixtures/generic.html#foo", {
+          onLoad: () ->
+            throw new Error("should not have called onLoad")
```

Comment:
this is already very clear and good but you get extra tokens for making this a `cy.stub()` and then asserting it was never called.

it makes the intention slightly more readable but better yet, had the stub **been** called the Cypress UI makes it easy to see when that happened.

You will find **plenty** of other cypress tests which don't do this, and its because they failed to earn their extra tokens